x/emissions/keeper/inference_synthesis/network_inference_builder.go

```diff
@@ -218,134 +337,244 @@ func (b *NetworkInferenceBuilder) SetOneOutInfererValues() *NetworkInferenceBuil
+	ctx sdk.Context,
+	k emissionskeeper.Keeper,
+	logger log.Logger,
+	topicId uint64,
+	allInferersAreNew bool,
+	inferers []Inferer,
+	infererToInference map[Inferer]*emissions.Inference,
+	forecasterToForecast map[Forecaster]*emissions.Forecast,
+	forecasterToForecastImpliedInference map[Forecaster]*emissions.Inference,
+	epsilonTopic alloraMath.Dec,
+	epsilonSafeDiv alloraMath.Dec,
+	pNorm alloraMath.Dec,
+	cNorm alloraMath.Dec,
+	oneInForecaster Forecaster,
+) (
+	oneInInference alloraMath.Dec,
+	err error,
+) {
+	logger.Debug(fmt.Sprintf("Calculating one-in inference for forecaster: %s", oneInForecaster))
 
 	// In each loop, remove all forecast-implied inferences except one
-	forecastImpliedInferencesWithForecaster := make(map[Worker]*emissions.Inference)
-	forecastImpliedInferencesWithForecaster[oneInForecaster] = palette.ForecastImpliedInferenceByWorker[oneInForecaster]
-	palette.ForecastImpliedInferenceByWorker = forecastImpliedInferencesWithForecaster
+	singleForecastImpliedInference := make(map[Worker]*emissions.Inference, 1)
+	singleForecastImpliedInference[oneInForecaster] = forecasterToForecastImpliedInference[oneInForecaster]
 
 	// Get self regret for the forecaster
-	regret, _, err := palette.K.GetOneInForecasterNetworkRegret(palette.Ctx, palette.TopicId, oneInForecaster, oneInForecaster)
+	singleForecasterRegret := make(map[Worker]*Regret, 1)
+	regret, _, err := k.GetOneInForecasterNetworkRegret(ctx, topicId, oneInForecaster, oneInForecaster)
 	if err != nil {
 		return alloraMath.Dec{}, errorsmod.Wrapf(err, "Error getting one-in forecaster regret")
 	}
+	singleForecasterRegret[oneInForecaster] = &regret.Value
 
-	palette.ForecasterRegrets[oneInForecaster] = &regret.Value
+	// get self forecast list
+	singleForecaster := []Worker{oneInForecaster}
 
-	remainingForecaster := []Worker{oneInForecaster}
-	err = palette.UpdateForecastersInfo(remainingForecaster)
-	if err != nil {
-		return alloraMath.Dec{}, errorsmod.Wrapf(err, "Error updating forecasters")
-	}
+	// get map of Forecaster to their forecast for the single forecaster
+	singleForecasterToForecast := make(map[Forecaster]*emissions.Forecast, 1)
+	singleForecasterToForecast[oneInForecaster] = forecasterToForecast[oneInForecaster]
 
 	// Get one-in regrets for the forecaster and the inferers they provided forecasts for
-	for _, inferer := range palette.Inferers {
-		regret, _, err := palette.K.GetOneInForecasterNetworkRegret(palette.Ctx, palette.TopicId, oneInForecaster, inferer)
+	infererToRegretForSingleForecaster := make(map[Inferer]*Regret)
+	infererToInferenceForSingleForecaster := make(map[Inferer]*emissions.Inference)
+	for _, inferer := range inferers {
+		regret, _, err := k.GetOneInForecasterNetworkRegret(ctx, topicId, oneInForecaster, inferer)
 		if err != nil {
-			return alloraMath.Dec{}, errorsmod.Wrapf(err, "Error getting one-in forecaster regret")
+			return alloraMath.Dec{}, errorsmod.Wrapf(err, "Calc one in value error getting one-in forecaster regret")
```

Comment:
Error message out of context

Reply:
what's the issue here? I think the error message is correct?

Reply:
Here we are getting one-in regrets, not calculating

Reply:
yeah we calc one-in inferences, not regrets. The regrets we just fetch.

Reply:
ah ok! sorry just double-checked it, I'm good with this! But just make sure to follow the same pattern in other functions